server/Dockerfile

```diff
@@ -6,11 +6,11 @@ FROM node:16-alpine
 
 WORKDIR /app
 
-ARG SERVER_SKIP_CI
+ARG SERVER_SKIP_NPM_CI
 
 COPY --chown=node:node . .
 
-RUN if [ -z "$SERVER_SKIP_CI" ] || [ $SERVER_SKIP_CI != 1 ]; then npm ci; fi
+RUN if [ -z "$SERVER_SKIP_NPM_CI" ] || [ $SERVER_SKIP_NPM_CI != 1 ]; then npm ci; fi

```

Comment:
instead of using the `SERVER_RUN_TESTS` and `SERVER_SKIP_NPM_CI` to check if we are running tests , could we just utilize the `NODE_ENV` here, like `NODE_ENV: "test"`. Other backend functionality uses this to determine the running environment.

Something like:
* `CMD ["/bin/sh", "-c", "npm run migration:up && npm run seed:up && if [ $NODE_ENV == "test"]; then npm test; else npm run start; fi"]
`
* `RUN if [ -z "$NODE_ENV" ] || [ $NODE_ENV != "test" ]; then npm ci; fi`

Reply:
For `SERVER_RUN_TESTS` that should work, but `SERVER_SKIP_NPM_CI` is purely build time functionality unrelated to whether tests are being run. It's a convenience for the developer so they don't need to wait for the dependencies to install every time they build the backend container.

Reply:
Thanks for the update and clarification